### ISSUE TITLE
renderSwatchLegend

### ISSUE THREAD

**Issue by creold** [schroef/Illustrator-Scripts#5]
Unobvious behavior of the script. It forcibly adds a string of color values in the current color model of the document, while we can also select the same color with a checkbox. Because of this, the information is duplicated =)
![Screenshot](https://github.com/schroef/Illustrator-Scripts/assets/32217031/39a36ad6-9e6e-4301-ad4f-fa014500fd41)

**Comment by schroef** [comment]
I was already thinking of that now when you mentioned this "issue" ;)

Currently working on an idea to do gradients. Buts its a huge hustle converting this script. Perhaps i just need to re-work it. In it current form running over the color-stops is hardly impossible for it to run well

**Comment by creold** [comment]
I agree with your thought. There is a lot of legacy code accumulated and maybe a full refactoring is needed to add a loop of gradient color stops and they are parsed by a common function that gets color components.

**Comment by schroef** [comment]
I did have another script called jd-export, this has a much smaller function for the loop. But its limited in the out out. I have already added the gradient colors to that this week. I think I can have a test using that function and combining it in this script

The script outputs a CSV file l, with per row all the color components

It's also in this repo, I just added it

**Comment by schroef** [comment]
@creold 

I added your idea for the color name as an option. See this commit; ad9ee654f00b929e4bb775d091c59df22fec5ddb

**Comment by creold** [comment]
Thank you. Now I don't get confused with channel values when the default swatch name is also made up of color values 😁

**Comment by schroef** [comment]
Nice!

Ps i checked your GitHub and you have such great tools and love how well written all the documentation is.
I guess you also do logo design. I made this tool, actually upgraded some else's basic tool, to a full logo export panel for illustrator. It will automatically export all the logos, 4 types, into all the needed formats. It's does this in seconds. I used to do this manually and it takes couple hours sometimes. It's a bit like the paid version named logo package, this one is free though. Perhaps it comes in handy for you

**Comment by creold** [comment]
Thanks for reviewing my repositories on Github. I remember the commercials for the logo package extension. I saw your "Logo Packer" for the first time, interesting development. I will keep the extension in mind for the future. I hardly make logos, I mostly draw illustrations for digital services.

**Comment by schroef** [comment]
Ah I see, I did notice you do lots of illustrations. Than I guess the default export for screens works good enough

**Comment by ELENAMEDICHI3246** [comment]
<img width="620" height="186" alt="Image" src="https://github.com/user-attachments/assets/c3b59a89-3c90-4f05-8be6-ba9e92ea1467" /> for some reason, this error appeared in the program window.
